ui/dashboards/src/components/PanelDrawer/PanelDrawer.tsx

```diff
@@ -184,17 +194,20 @@ const PanelDrawer = () => {
             <FormControl>
               <InputLabel id="panel-type-label">Panel Type</InputLabel>
               <Select required labelId="panel-type-label" label="Panel Type" value={kind} onChange={handleKindChange}>
-                {/* TODO: Replace this with options that come from asking the plugin system what panel plugins are available */}
-                <MenuItem value="LineChart">Line Chart</MenuItem>
-                <MenuItem value="GaugeChart">Gauge Chart</MenuItem>
-                <MenuItem value="StatChart">Stat Chart</MenuItem>
+                {pluginMetadata?.map((metadata) => (
+                  <MenuItem key={metadata.kind} value={metadata.kind}>
+                    {metadata.display.name}
+                  </MenuItem>
+                ))}
               </Select>
             </FormControl>
           </Grid>
           <Grid item xs={8}>
-            <PluginBoundary loadingFallback="Loading..." ErrorFallbackComponent={ErrorAlert}>
-              {kind !== '' && <PanelOptionsEditor kind={kind} value={options} onChange={handleOptionsChange} />}
-            </PluginBoundary>
+            <ErrorBoundary FallbackComponent={ErrorAlert}>
+              {OptionsEditorComponent !== undefined && (
+                <OptionsEditorComponent value={options} onChange={handleOptionsChange} />
+              )}
```

Comment:
Part of me wants to keep PanelOptionsEditor because I'm imagining a dev who is developing their own UI for editing panels... (i.e. they wouldn't just reach for PanelDrawer, they would build it out and PanelOptionsEditor would be a piece of it)

But, maybe we do just want them to reach for PanelDrawer, and we can assume that that's their point of entry into the editing experience. 

Which is making me realize I want to rename it something like `PanelEditor` or `EditPanelDrawer` (something to make it clear that it's the key to editing panels)

Reply:
Yeah, I can see a use case for maybe bringing it back in some way just to have self-contained loading state and UI for when the plugin component is loading. There's nothing stopping someone who's building their own UI from doing the same thing we're doing either to get the component (i.e. `const plugin = usePlugin('Panel', panelPluginKind)`).